### PR TITLE
added context and functionality

### PR DESCRIPTION
**Pull request by swethaogeti**
added likes, history, watchlater , playlist functionality

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -4,14 +4,31 @@ import { Routes, Route } from "react-router-dom";
 import HomePage from "./pages/homepage/HomePage";
 import Login from "./pages/loginpage/LoginPage";
 import Signup from "./pages/signupPage/SignupPage";
+import History from "./pages/historyPage/History";
+import Playlist from "./pages/playlistPage/Playlist";
+import Watchlater from "./pages/watchlaterPage/Watchlater";
+import Likes from "./pages/likesPage/Likes";
+import SingleVideo from "./pages/singleVideoPage/SingleVideo";
+import PageNotFound from "./pages/pageNotFound/PageNotFound";
+import PlaylistVideos from "./pages/playlistVideos/PlaylistVideos";
```
Comment (sri-rishi) [inline src/App.js:13]: Suggestion: you can add index.js file in pages folder and import all files from there

File: src/components/sidebarRow/SidebarRow.js [hunk 2]
```diff
@@ -1,10 +1,19 @@
...
-const SidebarRow = ({ title, Icon, active }) => {
+const SidebarRow = ({ title, Icon, active, link }) => {
   return (
-    <div className={active ? "sidebarRow" : "resize"}>
-      <Icon className="sidebarRow-icon" />
-      <h2 className="sidebarRow-title">{title}</h2>
+    // <div className={active ? "sidebarRow" : "resize"}>
+    //   <Icon className="sidebarRow-icon" />
+    //   <h2 className="sidebarRow-title">{title}</h2>
+    // </div>
```
Comment (sri-rishi) [inline src/components/sidebarRow/SidebarRow.js:9]: Suggestion: remove comments if not used

File: src/context/HistoryProvider.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    const fetchHistory = async () => {
+      if (user.token) {
+        try {
+          const historyResponse = await axios.get("/api/user/history", {
+            headers: { authorization: user.token },
+          });
+
+          dispatch({ type: "History", payload: historyResponse.data.history });
+        } catch (err) {
+          console.log("error in history fetch", err.message);
```
Comment (sri-rishi) [inline src/context/HistoryProvider.js:24]: Suggestion: can use `console.error` for errors

**Review by sri-rishi — APPROVED** [review]
code looks good, feel free to merge

File: src/App.js (hunk 1 above)
Comment (atharvabhanage02) [inline src/App.js:13]: You can have a `index.js` for all the pages combined and then access them

File: src/components/horizontalCard/horizontalCard.css [hunk 4]
```diff
@@ -0,0 +1,32 @@
+.horizontalCard {
+  width: 330px;
```
Comment (atharvabhanage02) [inline src/components/horizontalCard/horizontalCard.css:2]: Can use `rem`

File: src/components/longHorizontalCard/longHorizontalCard.css [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+.longCard-info {
+  padding-left: 0.4rem;
+
+  display: flex;
+  flex-direction: column;
+  align-items: flex-start;
+}
+
+.longHorizontalCard:hover {
+  background-color: whitesmoke;
```
Comment (atharvabhanage02) [inline src/components/longHorizontalCard/longHorizontalCard.css:22]: Use color variables

File: src/components/modal/modal.css [hunk 6]
```diff
@@ -0,0 +1,88 @@
+.modalBg {
+  width: 100vw;
+  height: 100vh;
+  position: fixed;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  cursor: pointer;
+  z-index: 10;
+  background-color: #131313c5;
```
Comment (atharvabhanage02) [inline src/components/modal/modal.css:10]: User color variables

File: src/components/sidebar/sidebar.css [hunk 7]
```diff
@@ -1,3 +1,52 @@
...
+
+.sidebarRow-title {
+  /* flex: 1; */
+
+  font-size: 0.9rem;
+  font-weight: 500;
+}
+
+.sidebarRow:hover {
+  /* background-color: lightgray; */
```
Comment (atharvabhanage02) [inline src/components/sidebar/sidebar.css:27]: Can remove commended code before merging

File: src/context/WatchlaterProvider.js [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+        dispatch({ type: "Watchlater", payload: [] });
+      }
+    };
+
+    fetchWatchlater();
+  }, [user]);
+
+  const addVideoToWatchlater = async (video) => {
+    try {
+      const addVideoToWatchlaterResponse = await axios.post(
```
Comment (atharvabhanage02) [inline src/context/WatchlaterProvider.js:40]: You can create  `services.js` file and have all Services Functions there

**Review by atharvabhanage02 — APPROVED** [review]
- Code is clean and Well Structured
- Logic is well implemented

Suggested few changes below , else everything look Fine

**Pull request merged**